activesupport/lib/active_support/message_verifier.rb

```diff
@@ -64,54 +66,156 @@ def valid_message?(signed_message)
+    # BASE64URL(JWS Payload)).
+    #
+    # ==== Options
+    #
+    # * <tt>:expires_at</tt> - Expiry time of the signed message.
+    # * <tt>:expires_in</tt> - Time from now after which the message
+    #   will expire (e.g. 1.month).
+    # * <tt>:for</tt> - Purpose/Scope of the signed message (defaults to 'universal').
+    #
+    # ==== Examples
+    #
+    # Messages can be signed to ensure that the data hasn't been tampered with.
     #
     #   verifier = ActiveSupport::MessageVerifier.new 's3Krit'
-    #   verifier.generate 'a private message' # => "BAhJIhRwcml2YXRlLW1lc3NhZ2UGOgZFVA==--e2d724331ebdee96a10fb99b089508d1c72bd772"
-    def generate(value)
-      data = encode(@serializer.dump(value))
-      "#{data}--#{generate_digest(data)}"
+    #   verifier.generate('a private message')
+    #   # => "BAh7B0kiCHR5cAY6BkVUSSIISldUBjsAVEkiCGFsZwY7AFRJIglTSEExBjsAVA==.BAh7CDoIcGxkSSIWYSBwcml2YXRlIG1lc3NhZ2UGOgZFVDoIZm9ySSIOdW5pdmVyc2FsBjsGVDoIZXhwSSIdMjAxNS0wOC0wN1QxNjo0NDowNi4xOTBaBjsGVA==.MjM0OGQ1MzkyYjE5YWIzNDg5ZTExNjEyNzZlOTNmOThmOGIyYzFiOQ=="
+    #
+    # You can even bump the security up some more by explaining what purpose a
+    # message is for. In this way evildoers can't reuse a sign-up form's cookie
+    # on the login page. For example.
+    #
+    #   signup_message = verifier.generate('a private message', for: 'signup_form')
+    #   # => "BAh7B0kiCHR5cAY6BkVUSSIISldUBjsAVEkiCGFsZwY7AFRJIglTSEExBjsAVA==.BAh7CDoIcGxkSSIWYSBwcml2YXRlIG1lc3NhZ2UGOgZFVDoIZm9ySSIQc2lnbnVwX2Zvcm0GOwZUOghleHBJIh0yMDE1LTA4LTA3VDE2OjQ1OjIxLjc5OFoGOwZU.YjE2MmQzYmM5NDY4YWFkMTQ1ZGUxNzExZTE5NTUxMmViN2ZjYzcyYw=="
+    #
+    #   verifier.verify(signup_message, for: 'signup_form')
+    #   # => 'a private message'
+    #
+    # You can also generate messages that expire some time in future. This is
+    # useful for cookies/messages which shouldn't be indefinitely accessible.
+    # For example:
+    #
+    #   expiring_message = verifier.generate('a private message', for: 'remember_me', expires_in: 1.hour)
+    #   # => "BAh7B0kiCHR5cAY6BkVUSSIISldUBjsAVEkiCGFsZwY7AFRJIglTSEExBjsAVA==.BAh7CDoIcGxkSSIWYSBwcml2YXRlIG1lc3NhZ2UGOgZFVDoIZm9ySSIQcmVtZW1iZXJfbWUGOwZUOghleHBJIh0yMDE1LTA3LTA3VDE3OjQ2OjI0LjMxMVoGOwZU.ZjdmZTRkOTc5YzcwZTUxOWM1YmQ1YWY5Nzg2NGU0YWY4YzAxNjdkNg=="
+    #
+    #   # Within 1 hour...
+    #   verifier.verify(expiring_message, for: 'remember_me')
+    #   # => "a private message"
+    #
+    #   # After 1 hour...
+    #   verifier.verify(expiring_message, for: 'remember_me')
+    #   # => ActiveSupport::Claims::ExpiredClaims
+    #
+    # You can also use +expires_at+ instead of +expires_in+. Just replace +expires_in+ by +expires_at+
+    # and +1.hour+ by +1.hour.from_now+ in the above example.
+    def generate(value, options = {})
+      @claims = Claims.new(payload: value, **options)
+      data = [encode(serialized_header), encode(serialized_claims)].join('.')
+      "#{data}.#{encode(generate_digest(to_ascii(data)))}"
     end
```

Comment:
Yeah. They are using some other conversions. I can't figure out why as of now.
I think we can go just with ```::Base64.strict_encode64``` function instead of the one in this https://github.com/progrium/ruby-jwt/blob/master/lib/jwt.rb.